src/transformers/modeling_utils.py

```diff
@@ -3250,6 +3288,17 @@ def from_pretrained(
         if quantization_method_from_config == QuantizationMethod.GPTQ:
             model = quantizer.convert_model(model)
             model._is_quantized_training_enabled = True
+        elif quantization_method_from_config == QuantizationMethod.AWQ:
+            from .integrations import get_keys_to_not_convert, replace_with_awq_linear
+
+            modules_to_not_convert = get_keys_to_not_convert(model)
+
+            if quantization_config is None:
+                quantization_config = AwqConfig.from_dict(config.quantization_config)
+
+            model, _ = replace_with_awq_linear(
```

Comment:
Maybe make use of the `has_been_replace` flag ? Just like bnb, we can log a warning if we were unable to replace the linear. If not, let's remove that arg  from the function

Reply:
Great point!